Add Fleet/Agent 8.6.0 release notes (#2450)
elastic/observability-docs: docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc

This change only affects standalone agents. Unique IDs are automatically generated in
agent policies managed by {fleet}. For more information, refer to
{agent-pull}/1994[#1994]

*Impact* +
Make sure that your standalone agent policies have a unique ID.
====

[discrete]
[[breaking-1140]]
.Diagnostics `--pprof` argument has been removed and is now always provided
[%collapsible]
====
*Details* +
The `diagnostics` command gathers diagnostic information about the {agent} and
each component/unit it runs. Starting in 8.6.0, the `--pprof`
argument is no longer available because `pprof` information is now always
provided. For more information, refer to {agent-pull}1140[#1140].

*Impact* +
Remove the `--pprof` argument from any scripts or commands you use.
====

[discrete]
[[known-issues-8.6.0]]
=== Known issues

[discrete]
[[known-issue-issue-2066]]
.Osquery live query results can take up to five minutes to show up in {kib}
[%collapsible]
====
*Details* +
A known issue in {agent} may prevent live query results from being available
in the {kib} UI even though the results have been successfully sent to {es}. 
For more information, refer to {agent-issue}2066[#2066].

*Impact* +

> Be aware that the live query results shown in {kib} may be stale.

Reviewer: ```suggestion
Be aware that the live query results shown in {kib} may be delayed by up to 5 minutes.
```

We can be more specific about the impact, we know that the results will be up to 5 minutes late because it is tied to the agent checkin period with the same duration.

====

[discrete]
[[new-features-8.6.0]]
=== New features

The 8.6.0 release adds the following new and notable features.

{fleet}::
* Differentiate kubernetes integration multipage experience {kibana-pull}145224[#145224]
* Add prerelease toggle to Integrations list {kibana-pull}143853[#143853]
* Add link to allow users to skip multistep add integration workflow {kibana-pull}143279[#143279]
* Add new action for single agents to request diagnostics {kibana-pull}142369[#142369]

{agent}::
* Upgrade Node to version 18.12.0 {agent-pull}1657[#1657] 
* Add experimental support for running the elastic-agent-shipper {agent-pull}1527[#1527] {agent-issue}219[#219]

> * Capture stdout/stderr of all subprocesses and adjust default log level to INFO for all components {agent-pull}1702[#1702] {agent-issue}221[#221]

Reviewer 2: The wording has changed for this one:
`Collect logs from sub-processes via stdout and stderr and write them to a single, unified Elastic Agent log file.`

* Remove inputs when all streams are removed {agent-pull}1869[#1869] {agent-issue}1868[#1868]
* No longer restart {agent} on log level change {agent-pull}1914[#1914] {agent-issue}1896[#1896]
* Add `inspect components` command to inspect the computed components/units model of the current configuration (for example, `elastic-agent inspect components`) {agent-pull}1701[#1701] {agent-issue}836[#836]
* Add support for the Common Expression Language (CEL) {filebeat} input type {agent-pull}1719[#1719]
* Only support {es} as an output for the beta synthetics integration {agent-pull}1491[#1491]

[discrete]
[[enhancements-8.6.0]]
=== Enhancements

{fleet}::
* Add `?full` option to get package info endpoint to return all package fields {kibana-pull}144343[#144343]

{agent}::

> * Health Status: {agent} now indicates detailed status information for each component/unit {fleet-server-pull}1747[#1747] {agent-issue}100[#100]

Reviewer: ```suggestion
* Health Status: {agent} now indicates detailed status information for each sub-process and input type {fleet-server-pull}1747[#1747] {agent-issue}100[#100]
```

We haven't defined component or unit anywhere so let's avoid use clearer language.

[discrete]
[[bug-fixes-8.6.0]]
=== Bug fixes

> {fleet}::

Reviewer: We should include https://github.com/elastic/fleet-server/issues/2205 as a bug fix.

The changelog entry can be something like "Fix synchronization bug that can lead to Elasticsearch being flooded by requests to /.fleet-actions/_fleet/_fleet_search."

> * Only show {fleet}-managed data streams on data streams list page {kibana-pull}143300[#143300]

Reviewer: ```suggestion
* Only show {fleet}-managed data streams on data streams list page {kibana-pull}143300[#143300]
* Fix synchronization bug in {fleet-server} that can lead to Elasticsearch being flooded by requests to /.fleet-actions/_fleet/_fleet_search {fleet-server-pull}2205[#2206].
```

We should include https://github.com/elastic/fleet-server/issues/2205 as a bug fix.

Author: Added a couple of minor edits to this suggestion before applying.

* No longer restart {agent} on log level change {agent-pull}1914[#1914] {agent-issue}1896[#1896]
* Add `inspect components` command to inspect the computed components/units model of the current configuration (for example, `elastic-agent inspect components`) {agent-pull}1701[#1701] {agent-issue}836[#836]
* Add support for the Common Expression Language (CEL) {filebeat} input type {agent-pull}1719[#1719]
* Only support {es} as an output for the beta synthetics integration {agent-pull}1491[#1491]

[discrete]
[[enhancements-8.6.0]]
=== Enhancements

{fleet}::
* Add `?full` option to get package info endpoint to return all package fields {kibana-pull}144343[#144343]

{agent}::
* Health Status: {agent} now indicates detailed status information for each sub-process and input type {fleet-server-pull}1747[#1747] {agent-issue}100[#100]

[discrete]
[[bug-fixes-8.6.0]]
=== Bug fixes

{fleet}::
* Only show {fleet}-managed data streams on data streams list page {kibana-pull}143300[#143300]
* Fix synchronization bug in {fleet-server} that can lead to {es} being flooded by requests to `/.fleet-actions/_fleet/_fleet_search` {fleet-server-pull}2205[#2206].

{agent}::

> * {agent} uses local port when running {fleet-server} {agent-pull}1867[#1867]

Author: ```suggestion
* {agent} now uses the locally bound port (8221) when running {fleet-server} instead of the external port (8220 {agent-pull}1867[#1867]
```

> * {agent} now uses the locally bound port (8221) when running {fleet-server} instead of the external port (8220 {agent-pull}1867[#1867]

Author: ```suggestion
* {agent} now uses the locally bound port (8221) when running {fleet-server} instead of the external port (8220 {agent-pull}1867[#1867]
```

> * {agent} now uses the locally bound port (8221) when running {fleet-server} instead of the external port (8220 {agent-pull}1867[#1867]

Author: ```suggestion
* {agent} now uses the locally bound port (8221) when running {fleet-server} instead of the external port (8220 {agent-pull}1867[#1867]
```

> * {agent} now uses the locally bound port (8221) when running {fleet-server} instead of the external port (8220 {agent-pull}1867[#1867]

Author: ```suggestion
* {agent} now uses the locally bound port (8221) when running {fleet-server} instead of the external port (8220) {agent-pull}1867[#1867]
```

* Add prerelease toggle to Integrations list {kibana-pull}143853[#143853]
* Add link to allow users to skip multistep add integration workflow {kibana-pull}143279[#143279]
* Add new action for single agents to request diagnostics {kibana-pull}142369[#142369]

{agent}::
* Upgrade Node to version 18.12.0 {agent-pull}1657[#1657] 
* Add experimental support for running the elastic-agent-shipper {agent-pull}1527[#1527] {agent-issue}219[#219]
* Collect logs from sub-processes via stdout and stderr and write them to a single, unified Elastic Agent log file {agent-pull}1702[#1702] {agent-issue}221[#221]
* Remove inputs when all streams are removed {agent-pull}1869[#1869] {agent-issue}1868[#1868]
* No longer restart {agent} on log level change {agent-pull}1914[#1914] {agent-issue}1896[#1896]
* Add `inspect components` command to inspect the computed components/units model of the current configuration (for example, `elastic-agent inspect components`) {agent-pull}1701[#1701] {agent-issue}836[#836]
* Add support for the Common Expression Language (CEL) {filebeat} input type {agent-pull}1719[#1719]
* Only support {es} as an output for the beta synthetics integration {agent-pull}1491[#1491]

> * New control protocol between the {agent} and its subprocesses enables per integration health reporting and simplifies new input development {agent-issue}836[#836] {agent-pull}1701[#1701]

Author: @pierrehilbert I've added the missing changelog items here. However, the change to the internal directory structure does not seem like a "feature" to me. Should it be covered under breaking changes instead?

Author: At this point, we should get this content merged and do follow-up PRs if additional changes are required. I just need approval....

Reviewer 2: I wanted to put it into "upgrade" kind but seems to be not working that's why I changed to "feature". But yes "breaking-change" makes probably more sense.

Reviewer: The internal directory structure change shouldn't be a breaking change, in that the user shouldn't need to take any action to account for it. Agent should keep working. It is definitely notable though.
